base/server/src/main/java/com/netscape/cmscore/dbs/LDAPPagedSearch.java

```diff
@@ -117,7 +119,7 @@ public List<CertRecord> getPage(int size)
                     LDAPv3.SCOPE_ONE, ldapfilter, ldapattrs, false, cons);
             DBSearchResults sr = new DBSearchResults(registry, res);
             while (sr.hasMoreElements()) {
-                entries.add((CertRecord) sr.nextElement());
+                entries.add(contentClassType.cast(sr.nextElement()));
             }
```

Comment:
Suppose we can use generics in `DBSearchResults` as well, we might not need the `contentClassType`.
```
ArrayList<E> entries = new ArrayList<>();
DBSearchResults<E> sr = new DBSearchResults<>(registry, res);
entries.add(sr.nextElement());
```

Reply:
I tried this approach but eclipse and java was complaining because `add()` is expecting an object from class E. With generic you cannot do dynamic type cast either `entries.add((E) sr.nextElement())`. The option I have investigated are:

- suppressing all the possible warning
- creating a custom class handling the the runtime type using reflection 
- provide the runtime type

I opted for the last one because it looks the cleaner and safer to implement but if we found better approaches we can always improve.